support/ci/build_component.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -eou pipefail
+
+component=${1?component argument required}
+
+# Since we are only verifying we don't have build failures, make everything
+# temp!
+export HAB_ORIGIN
+HAB_ORIGIN=throwaway
+# let's make a selfcontained tempdir for this job
+export JOB_TEMP_ROOT
+JOB_TEMP_ROOT=$(mktemp -d /tmp/job-root-XXXXXX)
+export HAB_CACHE_KEY_PATH
+HAB_CACHE_KEY_PATH="$JOB_TEMP_ROOT/keys"
```

Comment:
Ditto here, but not for `JOB_TEMP_ROOT` because of the possibility that any `$(…)` could fail
```bash
export HAB_CACHE_KEY_PATH="$JOB_TEMP_ROOT/keys"
```